src/components/EditorsColumn.jsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import isEmpty from 'lodash/isEmpty';
+import includes from 'lodash/includes';
+import partial from 'lodash/partial';
+
+import EditorContainer from './EditorContainer';
+import Editor from './Editor';
+
+class EditorsColumn extends React.Component {
+  _allErrorsFor(language) {
+    if (language === 'javascript') {
+      return this.props.errors.javascript.items.
+      concat(this.props.runtimeErrors);
+    }
+
+    return this.props.errors[language].items;
+  }
+
+  render() {
+    const {
+      currentProject,
+      onComponentMinimize,
+      onEditorInput,
+      onRequestedLineFocused,
+      ui,
+    } = this.props;
+    const editors = [];
+    ['html', 'css', 'javascript'].forEach((language) => {
+      if (includes(ui.minimizedComponents, `editor.${language}`)) {
+        return;
+      }
+
+      editors.push(
+        <EditorContainer
+          key={language}
+          language={language}
+          source={currentProject.sources[language]}
+          onMinimize={
+            partial(onComponentMinimize, `editor.${language}`)
+          }
+        >
+          <Editor
+            errors={this._allErrorsFor(language)}
+            key={language}
+            language={language}
+            percentageOfHeight={1 / editors.length}
+            projectKey={currentProject.projectKey}
+            requestedFocusedLine={ui.editors.requestedFocusedLine}
+            source={currentProject.sources[language]}
+            onInput={partial(onEditorInput, language)}
+            onRequestedLineFocused={onRequestedLineFocused}
+          />
+        </EditorContainer>,
+      );
+    });
+
+    if (isEmpty(editors)) {
+      return null;
+    }
+
+    return (
+      <div className="environment__column">
+        <div className="environment__columnContents editors">{editors}</div>
+      </div>
+    );
+  }
+}
+
+EditorsColumn.propTypes = {
+  currentProject: React.PropTypes.object.isRequired,
+  errors: React.PropTypes.object.isRequired,
+  runtimeErrors: React.PropTypes.array.isRequired,
+  ui: React.PropTypes.object.isRequired,
```

Comment:
Would it make more sense to only pass the parts of the `ui` state that this component needs?